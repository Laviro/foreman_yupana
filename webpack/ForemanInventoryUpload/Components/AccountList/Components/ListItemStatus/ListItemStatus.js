import React from 'react';
import { Grid } from 'patternfly-react';
import PropTypes from 'prop-types';
import { translate as __ } from 'foremanReact/common/I18n';
import './listItemStatus.scss';
import { getStatusIconByRegex } from './ListItemStatusHelper';

const ListItemStatus = ({ statuses }) => {
  const generatingStatusIcon = getStatusIconByRegex(
    statuses.generate_report_status
  );
  const uploadingStatusIcon = getStatusIconByRegex(
    statuses.upload_report_status
  );
  return (
    <Grid className="status">
      <Grid.Col sm={6} className="item">
        <p>{__('Generating')}</p>
        {generatingStatusIcon}
      </Grid.Col>
      <Grid.Col sm={6} className="item">
        <p>{__('Uploading')}</p>
        {uploadingStatusIcon}
      </Grid.Col>
    </Grid>
  );
};

ListItemStatus.propTypes = {
  statuses: PropTypes.shape({
    generate_report_status: PropTypes.string,
    upload_report_status: PropTypes.string,
  }),
};

ListItemStatus.defaultProps = {
  statuses: {
    generate_report_status: 'unknown',
    uploupload_report_statusading: 'unknown',
  },
};

export default ListItemStatus;
