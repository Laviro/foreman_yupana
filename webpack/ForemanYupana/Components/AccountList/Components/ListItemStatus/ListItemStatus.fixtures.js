export const props = {
  statuses: {
    generate_report_status: 'Exit Code: pid 22191 exit 0',
    upload_report_status: 'Exit Code: Running in pid 22376',
  },
};
