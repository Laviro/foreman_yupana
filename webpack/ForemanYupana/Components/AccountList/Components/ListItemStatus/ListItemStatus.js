import React from 'react';
import { Grid, Icon, Spinner } from 'patternfly-react';
import PropTypes from 'prop-types';
import './listItemStatus.scss';

const ListItemStatus = ({ status }) => {
  const statusIcons = {
    success: <Icon name="check" />,
    failure: <Icon name="times" />,
    running: <Spinner loading inline size="xs" />,
    stopped: <span>--</span>,
  };
  return (
    <Grid className="status">
      <Grid.Col sm={6} className="item">
        <p>Generating</p>
        {statusIcons[status.generating]}
      </Grid.Col>
      <Grid.Col sm={6} className="item">
        <p>Uploading</p>
        {statusIcons[status.uploading]}
      </Grid.Col>
    </Grid>
  );
};

ListItemStatus.propTypes = {
  status: PropTypes.shape({
    generating: PropTypes.string,
    uploading: PropTypes.string,
  }),
};

ListItemStatus.defaultProps = {
  status: {
    generating: 'stopped',
    uploading: 'stopped',
  },
};

export default ListItemStatus;
