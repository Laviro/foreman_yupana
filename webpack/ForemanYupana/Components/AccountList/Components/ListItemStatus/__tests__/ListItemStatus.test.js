import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import ListItemStatus from '../ListItemStatus';

const fixtures = {
  'render without Props': {},
  /** fixtures, props for the component */
};

describe('ListItemStatus', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(ListItemStatus, fixtures));
});
