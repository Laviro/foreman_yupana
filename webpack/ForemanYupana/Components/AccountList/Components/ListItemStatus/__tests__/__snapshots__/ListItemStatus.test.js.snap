// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemStatus rendering render without Props 1`] = `
<Grid
  bsClass="container"
  className="status"
  componentClass="div"
  fluid={false}
>
  <Col
    bsClass="col"
    className="item"
    componentClass="div"
    sm={6}
  >
    <p>
      Generating
    </p>
    <span>
      --
    </span>
  </Col>
  <Col
    bsClass="col"
    className="item"
    componentClass="div"
    sm={6}
  >
    <p>
      Uploading
    </p>
    <span>
      --
    </span>
  </Col>
</Grid>
`;
